sql/src/main/java/io/crate/executor/transport/RepositoryService.java

```
@@ -141,7 +139,9 @@ private void dropIfExists(String repoName, Runnable callback) {
         if (repository == null) {
             callback.run();
         } else {
-            execute(new DropRepositoryAnalyzedStatement(repoName)).addListener(callback, MoreExecutors.directExecutor());
+            execute(new DropRepositoryAnalyzedStatement(repoName)).whenComplete(
+                (r, e) -> callback.run()
```

isn't this swallowing exceptions?